enhancements/authentication/AuthConfig-missing-fields.md

```diff
@@ -0,0 +1,449 @@
+**Scenario:**  
+A cluster administrator needs to configure OpenShift to validate the audience (`aud`) claim in JWT tokens using a flexible matching policy.
+
+**Steps:**
+1. Update the authentication CRD with `AudienceMatchPolicy`:
+```yaml
+  apiVersion: config.openshift.io/v1
+  kind: Authentication
+  metadata:
+    name: cluster
+  spec:
+    type: OIDC
+    oidcProviders:
+    - name: foo
+      issuer:
+        audienceMatchPolicy: MatchAny
+```
+
+#### 5. Enforcing Token Expiration Limits (Claim Validation)  
+**Scenario:**  
+A security-conscious organization wants to ensure that all OIDC tokens used for authentication have a maximum expiration time of 24 hours.  
+
+**Steps:**  
+1. Update the authentication CRD to enforce a maximum token expiration time:  
+   ```yaml
+   apiVersion: config.openshift.io/v1
+   kind: Authentication
+   metadata:
+     name: cluster
+   spec:
+    type: OIDC
+    oidcProviders:
+      claimValidationRules:
+     - expression: 'claims.exp - claims.nbf <= 86400'
+        message: total token lifetime must not exceed 24 hours
+  ```
+
+#### 6. Restricting Reserved Usernames (User Validation)  
+**Scenario:**  
+An OpenShift administrator wants to prevent users from being created with reserved system prefixes, such as `system:`, to avoid conflicts with system users.  
+
+**Steps:**  
+1. Update the authentication CRD to enforce username restrictions:  
+   ```yaml
+  apiVersion: config.openshift.io/v1
+  kind: Authentication
+  metadata:
+    name: cluster
+  spec:
+    type: OIDC
+    oidcProviders:
+      userValidationRules:
+      - expression: "!user.username.startsWith('system:')" # the expression will evaluate to true, so validation will succeed.
+          message: 'username cannot used reserved system: prefix'
+  ```
+
+### API Extensions
+To facilitate the configuration and validation of token claims, token issuers, and user validation rules, the existing `authentications.config.openshift.io` CRD is extended with new fields and structures. These extensions provide enhanced flexibility in token validation, token claim mappings, issuer configuration, and user validation. The proposed changes introduce new fields that allow administrators to define custom authentication behaviors.  
+The proposed API changes have been submitted in [openshift/api#2245](https://github.com/openshift/api/pull/2245). Below are examples of how users will configure these new fields:  
+
+
+#### TokenIssuer  
+##### DiscoveryURL
+- **What it does:**  
+  The `discoveryURL` field specifies the OpenID Connect (OIDC) discovery endpoint, which provides metadata about the identity provider. It allows OpenShift to automatically retrieve issuer details, supported authentication methods, and token endpoints.  
+
+- **Why users would set it:**  
+  This field is necessary when integrating OpenShift authentication with an external OIDC provider, ensuring proper token validation.  
```

Comment:
This line makes it sound like this would be introduced as a required field but it is being introduced as an optional field.

IIUC this field is really only needed for the use case of a user using an OIDC provider that may not use the standard OIDC discovery endpoint (for whatever reason).

We should clarify that this isn't a field that would be set in any case but ^.